### ISSUE TITLE
All labels only scroll 3 times

### ISSUE THREAD

**Issue by abkodi** [xbmc/xbmc#15758]
ALL labels are scrolling its text as they should (3 times, then it stops), but AFTER this the label don't scroll anymore at all even not the content is changing (i.e. next song is playing, different content).
And this happens with ALL labels, in short: first time they are scrolling, but after then anymore at all. 

Used Operating system:
 - [X] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [X] Raspberry-Pi
 - [X] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: 18.0 / 18.1 / 18.2 RC

**Comment by HitcherUK** [comment]
Can you change the title to something more descriptive?

eg 'Labels only scroll 3 times'

Thanks.

**Comment by KarellenX** [comment]
Similar issue solved by disabling "smartredraw"

https://forum.kodi.tv/showthread.php?tid=344040

**Comment by HitcherUK** [comment]
> Similar issue solved by disabling "smartredraw"
> 
> https://forum.kodi.tv/showthread.php?tid=344040

Is it enabled by default then?

**Comment by peak3d** [comment]
No

**Comment by DaVukovic** [comment]
> And this happens with ALL labels

I would say this is not correct. Labels which are use for skins scroll 3 times if they are focused. Then, after 3 times the labels do stop scrolling as expected. If you unfocus and focus them again, they start scrolling. So it's not correct that this happens with "all labels"

**Comment by KarellenX** [comment]
> Is it enabled by default then?

No, as peak3d stated. The member had added it to the advancedsettings.xml

**Comment by HitcherUK** [comment]
> No, as peak3d stated. The member had added it to the advancedsettings.xml

Then it has nothing to do with this problem as this happens without it.

**Comment by KarellenX** [comment]
> Then it has nothing to do with this problem as this happens without it.

Ok, I only followed your link back here. If it has nothing to do with it, why is it referenced?

> @peak3d I think this one might related to #12213


Anyway, not important.

**Comment by peak3d** [comment]
If wanted I could make the 3-times scroll behaviour depend on smartredraw.

continuous scrolling labels lead to the fact that kodi GUI is recalculated / redrawn completely on every vsync. This means with kodi open (and scrolling label) your system will never go into low power mode.
This was the reason I implemented this feature.

**Comment by HitcherUK** [comment]
> Ok, I only followed your link back here. If it has nothing to do with smartredraw, why is it referenced?

This PR caused it but it doesn't matter if smartredraw is enabled or not is what I meant.

> If wanted I could make the 3-times scroll behaviour depend on smartredraw.
> 
> continuous scrolling labels lead to the fact that kodi GUI is recalculated / redrawn completely on every vsync. This means with kodi open (and scrolling label) your system will never go into low power mode.
> This was the reason I implemented this feature.

I get what you're trying to implement but if a skinner wants scrolling labels in their skin then that's what should happen. Linking it to smartredraw and giving the user the option of changing this behaviour sounds like a good compromise though.

**Comment by rwyarbrough** [comment]
If the object is to reduce the load on the cpu - instead of forcing a default of 3 scrolls and stop or using smartredraw to limit the number of scrolls to 3 cycles, maybe you could just add a variable that can be added to control number of scrolls inside the label control itself. 0=infinite (default for true), 100=100 scrolls - example: <scroll cycles="100">true</scroll>.

I'm sure there are plenty of people who would rather live with the increased cpu and have infinite scrolling versus losing control over the number of cycles.

**Comment by DevoDave** [comment]
@peak3d making it dependent on smartredraw would work for most, but having a bit more control like @rwyarbrough suggests above would be better IMHO.

**Comment by jurialmunkey** [comment]
Commenting to deal with the stale bot. This issue still exists in Omega onwards.
